src/lib/encoder/Encoder.h

```diff
@@ -99,14 +100,34 @@ inline unsigned Encoder::encode(const std::string& filename, std::string output_
 	return i;
 }
 
-inline unsigned Encoder::encode_fountain(const std::string& filename, const std::function<bool(const cv::Mat&, unsigned)>& on_frame)
+inline unsigned Encoder::encode_fountain(const std::string& filename, const std::function<bool(const cv::Mat&, unsigned)>& on_frame, int compression_level)
 {
-	std::ifstream f(filename);
-	fountain_encoder_stream fes = fountain_encoder_stream::create(f, cimbar::Config::fountain_chunk_size(_eccBytes), 0); // will eventually do something clever with encode_id?
+	unsigned chunk_size = cimbar::Config::fountain_chunk_size(_eccBytes);
+
+	std::stringstream ss;
+	std::ifstream infile(filename);
+	if (compression_level <= 0)
+		ss << infile.rdbuf();
+	else
+	{
+		cimbar::zstd_compressor<std::stringstream> f;
+		if (!f.compress(infile))
+			return 0;
+
+		// find size of compressed zstd stream, and pad it if necessary.
+		size_t compressedSize = f.size();
+		if (compressedSize < chunk_size)
+			f.pad(chunk_size - compressedSize + 1);
```

Comment:
wirehair wants the input data to need at least two chunks, and we (at least for now) would like to always use wirehair for simplicity's sake. So... pad it??